contracts/RiskFund/RiskFund.sol

```diff
@@ -141,16 +141,18 @@ contract RiskFund is Ownable2StepUpgradeable, ExponentialNoError, ReserveHelpers
         );
         require(canSwapPoolsAsset, "Risk fund: Not authorized to swap pool assets.");
         require(poolRegistry != address(0), "Risk fund: Invalid pool registry.");
-        require(
-            underlyingAssets.length == amountsOutMin.length,
-            "Risk fund: underlyingAssets and amountsOutMin are unequal lengths"
-        );
+        require(markets.length == amountsOutMin.length, "Risk fund: markets and amountsOutMin are unequal lengths");
 
         uint256 totalAmount;
-        uint256 underlyingAssetsCount = underlyingAssets.length;
-        for (uint256 i; i < underlyingAssetsCount; ++i) {
-            VToken vToken = VToken(underlyingAssets[i]);
+        uint256 marketsCount = markets.length;
+        for (uint256 i; i < marketsCount; ++i) {
+            VToken vToken = VToken(markets[i]);
             address comptroller = address(vToken.comptroller());
+
+            PoolRegistry.VenusPool memory pool = PoolRegistry(poolRegistry).getPoolByComptroller(comptroller);
+            require(pool.comptroller == comptroller, "comptroller doesn't exist pool registry");
+            require(Comptroller(comptroller).isMarketListed(vToken) == true, "market is not listed");
```

Comment:
```suggestion
            require(Comptroller(comptroller).isMarketListed(vToken), "market is not listed");
```
I think this would be enough, right?